### ISSUE TITLE
[10.3.0-rc2] Playlist not shown anymore

### ISSUE THREAD

**Issue by TwizzyDizzy** [jellyfin/jellyfin#1228]
**Describe the bug**
Playlist are not shown in UI anymore since upgrading to 10.3.0-rc2

**To Reproduce**
n/a

**Expected behavior**
Playlist should be shown under "My Media"

**Logs**
Playlist files are there:
```
root@pi:/var/lib/jellyfin/data/playlists# find ./ -ls
   276250      4 drwxr-xr-x   6 jellyfin jellyfin     4096 Mar 14 04:03 ./
   406725      4 drwxr-xr-x   2 jellyfin jellyfin     4096 Mar 14 04:03 ./emo-joint\ [playlist]
   406728      4 -rw-r--r--   1 jellyfin jellyfin      951 Apr 12 10:02 ./emo-joint\ [playlist]/playlist.xml
   658342      4 drwxr-xr-x   2 jellyfin jellyfin     4096 Mar 12 03:38 ./sunday-joint-interludes\ [playlist]
   658343      4 -rw-r--r--   1 jellyfin jellyfin     1303 Apr  9 21:58 ./sunday-joint-interludes\ [playlist]/playlist.xml
   406676      4 drwxr-xr-x   2 jellyfin jellyfin     4096 Mar 14 03:25 ./joint2\ [playlist]
   406677      4 -rw-r--r--   1 jellyfin jellyfin     1687 Apr 12 10:02 ./joint2\ [playlist]/playlist.xml
   783341      4 drwxr-xr-x   2 jellyfin jellyfin     4096 Mar 12 01:02 ./sunday-joint\ [playlist]
   783343      8 -rw-r--r--   1 jellyfin jellyfin     5696 Apr 11 10:01 ./sunday-joint\ [playlist]/playlist.xml
```

**Screenshots**
![image](https://user-images.githubusercontent.com/179595/56028947-99f44980-5d19-11e9-8dd4-f74534315b70.png)

**System (please complete the following information):**
 - OS: Raspbian Stretch (9, latest)
 - Browser: Chromium
 - Jellyfin Version: 10.3.0-rc2
 - Reverse proxy: Apache 2.4

**Additional context**
* Might relate to #1107, but I don't think so.
* I'm using Jellyfin behind an Apache 2.4 reverse-proxy

**Comment by TwizzyDizzy** [comment]
FYI: the actual request for the playlists fires, is answered by jellyfin with a status code of `200`...

```
Request URL: https://myhost/jellyfin/emby/Users/3435443c01e94e5c84bbbc7f50490cd1/Items?SortBy=SortName&SortOrder=Ascending&IncludeItemTypes=Playlist&Recursive=true&Fields=PrimaryImageAspectRatio%2CSortName%2CCanDelete&StartIndex=0&ParentId=47744e79ba0def9fdddacea5922e2d9e
Request Method: GET
Status Code: 200 OK
Remote Address: *.*.*.*:443
Referrer Policy: no-referrer-when-downgrade
```

Content returned:
```
{Items: [], TotalRecordCount: 0}
Items: []
TotalRecordCount: 0
```

**Comment by TwizzyDizzy** [comment]
FYI 2:

I just tried creating a new playlist. I added a song from an album to a playlist. The "New playlist" dialogue didn't show the old playlists, so I created a new one.

This one got created and all playlists are now shown again.

Don't know if this helps or further obscurs the issue, but I though I might just add it.

Cheers
Thomas

**Comment by TwizzyDizzy** [comment]
Now THAT is interesting. After deleting the new test playlist all went back to the behaviour before.

Next test was: create two new playlists and delete them one by one. After deleting the first, behaviour was OK (all playlists were shown) and only after deleting the second test-playlist things went back to false behaviour. This leads me to believe the actual count of my playlists is relevant... as you can see in the first post: there's 4 playlists.

Cheers
Thomas

**Comment by TwizzyDizzy** [comment]
Dafuq. Now my 4 test playlists re-appeared. This is getting more obscure every minute :D

**Comment by joshuaboniface** [comment]
So the biggest thing I can think of is that you have some weird JS or resources cached in the browser. Try removing all browser history for your JF instance then force-refreshing the page. Also if you hvaen't yet, enable Debug logging and let's see what it reports when they don't show up.

**Comment by TwizzyDizzy** [comment]
We can discard the possibility of weird caching-issues as I have completely disabled the cache (developer console) while verifying this issue. It's not that.

BUT: this indeed does not seem to be an issue of playlists but of my Apache reverse-proxying setup involving failed websocket requests as shown below. I'll open a separate issue, if it warrants for one.

```
Request URL: wss://myhost/jellyfin/embywebsocket?api_key=4c7c9ca9e1054992a172ba1a7ab6dba9&deviceId=TW96aWxsYS81LjAgKFgxMTsgTGludXggeDg2XzY0KSBBcHBsZVdlYktpdC81MzcuMzYgKEtIVE1MLCBsaWtlIEdlY2tvKSBDaHJvbWUvNzIuMC4zNjI2LjEyMiBTYWZhcmkvNTM3LjM2fDE1NTI1MjI5NzUyMDI1
Request Method: GET
Status Code: 404 Not Found
```

Jellyfin debug log:
```
Apr 19 13:42:02 myhost jellyfin[8696]: [13:42:02] [DBG] HTTP Response 404 to 192.168.1.1. Time: 0:00:00.0132773. http://myhost/embywebsocket?deviceId=TW96aWxsYS81LjAgKFgxMTsgTGludXggeDg2XzY0KSBBcHBsZVdlYktpdC81MzcuMzYgKEtIVE1MLCBsaWtlIEdlY2tvKSBDaHJvbWUvNzIuMC4zNjI2LjEyMiBTYWZhcmkvNTM3LjM2fDE1NTI1MjI5NzUyMDI1
```

**Comment by joshuaboniface** [comment]
Are you trying to add a subpath? I see a `/jellyfin` in the 404ing URL whoch would explain that. We don't yet support a configurable base path. If.you remove that do the websockets work?

**Comment by TwizzyDizzy** [comment]
Yes I do. I'm using jellyfin via Apache 2.4 reverse-proxy. `/jellyfin/` gets proxied to jellyfin `localhost:8096`.

Websockets do work when calling jellyfin directly (e.g. `http://jellyfin:8096`).

> We don't yet support a configurable base path.

I see. Thanks for that information, I suspected something like that. Is there an open issue I could subscribe to?

Cheers
Thomas

**Comment by joshuaboniface** [comment]
If you do get the rewrite-fu all working let me know and I'll add it to the docs! #691 is the feature issue to get this ssupported officially.

**Comment by TwizzyDizzy** [comment]
> If you do get the rewrite-fu all working let me know and I'll add it to the docs!

Seems like I got it working. Just tell me the page, I'll send a PR.

Cheers
Thomas

**Comment by TwizzyDizzy** [comment]
Despite the fact, that I got my websockets fixed I still have the feeling something is utterly wrong with playlists / playlist-management. Here's a general follow-up-issue: #1271.

**Comment by Bellusterra** [comment]
I’d like to add that in addition to the things mentioned above another very odd thing with the playlists is that if from the admin account an additional user is created, logging out and in to the added “guest” user account, create a playlist from the shared content from the admin account then the playlist tab is empty. Going back into the Admin account you find the playlist created and viewable created by the user account.

**Comment by Bellusterra** [comment]
Did some troubleshooting and found that the playlist appeared in the “guest” account only after giving that user “Enable access for all libraries” anything other than that and the guest accounts playlist comes up empty.
